presto-main/src/main/java/com/facebook/presto/execution/QueryStats.java

```diff
@@ -385,10 +385,13 @@ public static QueryStats create(
             operatorStatsSummary.addAll(stageExecutionStats.getOperatorSummaries());
             // We prepend each metric name with the stage id to avoid merging metrics across stages.
             int stageId = stageInfo.getStageId().getId();
-            stageExecutionStats.getRuntimeStats().getMetrics().values().forEach(metric -> {
-                String metricName = String.format("S%d-%s", stageId, metric.getName());
+            stageExecutionStats.getRuntimeStats().getMetrics().forEach((name, metric) -> {
+                String metricName = String.format("S%d-%s", stageId, name);
                 mergedRuntimeStats.getMetrics().computeIfAbsent(metricName, RuntimeMetric::new);
-                mergedRuntimeStats.getMetric(metricName).mergeWith(metric);
+                mergedRuntimeStats.getMetrics().computeIfPresent(metricName, (metricKey, metricValue) -> {
+                    metricValue.mergeWith(metric);
+                    return metricValue;
```

Comment:
We should not modify internal data structures outside the class. Lets add a new function `mergeMetricWith` which accepts the name and `RuntimeMetric` object and does this internally.

Reply:
Done.